homeassistant/components/esphome/light.py

```diff
@@ -414,11 +414,8 @@ def _on_static_info_update(self, static_info: EntityInfo) -> None:
 
         self._attr_supported_color_modes = supported
         self._attr_effect_list = static_info.effects
-        self._attr_min_mireds = round(static_info.min_mireds)
-        self._attr_max_mireds = round(static_info.max_mireds)
-        if ColorMode.COLOR_TEMP in supported:
-            self._attr_min_color_temp_kelvin = _mired_to_kelvin(static_info.max_mireds)
-            self._attr_max_color_temp_kelvin = _mired_to_kelvin(static_info.min_mireds)
+        self._attr_min_color_temp_kelvin = _mired_to_kelvin(static_info.max_mireds)
+        self._attr_max_color_temp_kelvin = _mired_to_kelvin(static_info.min_mireds)
```

Comment:
Note: I think the if check added in https://github.com/home-assistant/core/pull/91424 was a mistake

Reply:
Its probably ok as we have a divide by zero guard in `_mired_to_kelvin`

Reply:
It seems that if these were missing, the tests were somehow falling back to the previous min_mireds/max_mireds values.

So if you think it's the tests that are wrong, then I can tweak accordingly.

Reply:
I've double-checked again, and I'm confident the change is correct, and `if ColorMode.COLOR_TEMP in supported` is not needed